stake-pool/js/src/schema.ts

```diff
@@ -0,0 +1,181 @@
+import {Schema, serialize, deserializeUnchecked} from 'borsh';
+import BN from 'bn.js';
+
+// Class wrapping a plain object
+export abstract class Assignable {
+  constructor(properties: {[key: string]: any}) {
+    Object.keys(properties).forEach((key: string) => {
+      this[key] = properties[key];
+    });
+  }
+
+  encode(): Buffer {
+    return Buffer.from(serialize(SCHEMA, this));
+  }
+
+  static decode<T extends Assignable>(data: Buffer): T {
+    return deserializeUnchecked(SCHEMA, this, data);
+  }
+}
+
+// Class representing a Rust-compatible enum, since enums are only strings or
+// numbers in pure JS
+export abstract class Enum extends Assignable {
```

Comment:
You can remove this and use `Enum` from web3.js https://github.com/solana-labs/solana/blob/ec1a307a7c2618b9bdea7b82a39a36ba194ae0d0/web3.js/src/util/borsh-schema.ts#L21